server/src/test/java/org/elasticsearch/action/search/ReduceSearchPhaseExceptionTest.java

```diff
@@ -0,0 +1,167 @@
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.search.SearchShardTarget;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.hamcrest.Matchers.hasSize;
+
+public class ReduceSearchPhaseExceptionTest extends ESTestCase {
+
+    public void testToXContent() throws IOException {
+        ReduceSearchPhaseException exception = new ReduceSearchPhaseException("test", "all shards failed", new RuntimeException(),
+            new ShardSearchFailure[]{
+                new ShardSearchFailure(new ParsingException(1, 2, "foobar", null),
+                    new SearchShardTarget("node_1", new ShardId("foo", "_na_", 0), null, OriginalIndices.NONE)),
+                new ShardSearchFailure(new IndexShardClosedException(new ShardId("foo", "_na_", 1)),
+                    new SearchShardTarget("node_2", new ShardId("foo", "_na_", 1), null, OriginalIndices.NONE)),
+                new ShardSearchFailure(new ParsingException(5, 7, "foobar", null),
+                    new SearchShardTarget("node_3", new ShardId("foo", "_na_", 2), null, OriginalIndices.NONE)),
+            });
+
+        // Failures are grouped (by default)
+        final String expectedJson = XContentHelper.stripWhitespace(
+            "{"
+                + "  \"type\": \"reduce_search_phase_exception\","
+                + "  \"reason\": \"[reduce] all shards failed\","
+                + "  \"phase\": \"test\","
+                + "  \"grouped\": true,"
+                + "  \"failed_shards\": ["
+                + "    {"
+                + "      \"shard\": 0,"
+                + "      \"index\": \"foo\","
+                + "      \"node\": \"node_1\","
+                + "      \"reason\": {"
+                + "        \"type\": \"parsing_exception\","
+                + "        \"reason\": \"foobar\","
+                + "        \"line\": 1,"
+                + "        \"col\": 2"
+                + "      }"
+                + "    },"
+                + "    {"
+                + "      \"shard\": 1,"
+                + "      \"index\": \"foo\","
+                + "      \"node\": \"node_2\","
+                + "      \"reason\": {"
+                + "        \"type\": \"index_shard_closed_exception\","
+                + "        \"reason\": \"CurrentState[CLOSED] Closed\","
+                + "        \"index_uuid\": \"_na_\","
+                + "        \"shard\": \"1\","
+                + "        \"index\": \"foo\""
+                + "      }"
+                + "    }"
+                + "  ],"
+                + "  \"caused_by\": {"
+                + "                    \"type\": \"runtime_exception\","
+                + "                    \"reason\": null"
+                + "                }"
+                + "}"
+        );
+        System.out.println(Strings.toString(exception));
```

Comment:
Please remove System.out's from the test, they get flagged by our style checks as well.